### ISSUE TITLE
numeric string incorrectly considered a number in a property access

### ISSUE THREAD

**Issue by altendky** [rhysd/actionlint#250]
When accessing an attribute by indexing using a string object, the linter complains that the string is a number when the string is numeric.  This false positive can be avoided by adding non-numeric characters such as in https://github.com/Chia-Network/actions/pull/51/commits/5a9df1b1fbb0e473aeaf9a2feb4231548806627b.

https://github.com/Chia-Network/actions/actions/runs/3527762746/jobs/5917180165#step:4:76
```
------
.github/workflows/test-create-venv.yml:22:40: property access of object must be type of string but got "number" [expression]
   |
22 |     container: ${{ matrix.os.container[matrix.python.matrix] }}
   |                                        ^~~~~~~~~~~~~~~~~~~~~
.github/workflows/test-create-venv.yml:110:31: property access of object must be type of string but got "number" [expression]
    |
110 |       if: matrix.os.container[matrix.python.matrix] == ''
    |                               ^~~~~~~~~~~~~~~~~~~~~
------
```
